src/math/calculateLevelFromXP.ts

```diff
@@ -5,7 +5,7 @@ export default function calculateLevelFromXP(xp: number, game_id: Game): number
     const game_data = Games[game_id];
     if (!game_data) return null;
 
-    if (game_data.max_level === 0) return null;
+    if (game_data.max_level === 0 || !game_data.has_levels) return null;
```

Comment:
You don't need to check if `max_level` is equal to zero anymore, it's just a redundant check as a game with a max level of 0 doesn't have levels and should be marked as such
```suggestion
    if (!game_data.has_levels) return null;
```